src/app/auth/services/auth.service.ts

```diff
@@ -1,30 +1,66 @@
-import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, map, Observable, tap } from "rxjs";
+import { SessionStorageService } from "./session-storage.service";
+import { User, UserLogin } from "@app/models/user.model";
+import { Router } from "@angular/router";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class AuthService {
-    login(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  private isAuthorized$$ = new BehaviorSubject<boolean>(false);
+  public isAuthorized$ = this.isAuthorized$$.asObservable();
 
-    logout() {
-        // Add your code here
-    }
+  constructor(
+    private http: HttpClient,
+    private sessionStorageService: SessionStorageService,
+    private router: Router
+  ) {
+    this.isAuthorized$$.next(!!this.sessionStorageService.getToken());
+  }
 
-    register(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  login(user: UserLogin): Observable<any> {
+    return this.http.post("http://localhost:4000/login", user).pipe(
+      tap((response: any) => {
+        if (response.successful) {
+          this.sessionStorageService.setToken(response.result);
+          this.isAuthorized$$.next(true);
+          this.router.navigate(["/courses"]);
```

Comment:
Authentication or authorization services should not be responsible for navigation, only define access and level of permission.